src/Resources/js/views/mealIndex.ts

```diff
@@ -34,6 +34,64 @@ export default class MealIndexView {
         $('.meals-list .meal .meal-row').on('click', ' .title.edit', this.handleCombinedMealEdit.bind(this));
     }
 
+    private static handleUpdateParticipation(data: ParticipationCountData) {
+        $(`div[data-id=${data.mealId}] .count`).text(data.count);
+        if(data.isAvailable) {
+            $(`div[data-id=${data.mealId}] .participants-count`)
+                .removeClass('participation-limit-reached')
+                .addClass('participation-allowed');
+        } else {
+            $(`div[data-id=${data.mealId}] .participants-count`)
+                .removeClass('participation-allowed')
+                .addClass('participation-limit-reached');
+        }
+    }
+
+    private static handleUpdateOffers(data: OfferData) {
+        let $mealWrapper = $('[data-id=' + data.mealId + ']');
+        let $checkbox = $mealWrapper.find('.participation-checkbox');
+
+        // new offer available and checkbox not checked yet
+        if (data.isAvailable === true &&
+            $checkbox.is(':checked') === false &&
+            $checkbox.hasClass(ParticipationAction.UNSWAP) === false &&
+            $checkbox.hasClass(ParticipationAction.ACCEPT_OFFER) === false &&
+            $checkbox.hasClass(ParticipationAction.JOIN) === false) {
+            let date = data.date;
+            let dishSlug = data.dishSlug;
+            ParticipationUpdateHandler.changeToOfferIsAvailable(
+                $checkbox,
+                '/menu/' + date + '/' + dishSlug + '/accept-offer'
+            );
+        }
+
+        // if a user's offer is gone and the participation-badge is still showing 'pending', disable the checkbox, tooltip and change badge
+        if ($checkbox.hasClass(ParticipationAction.UNSWAP) === true) {
+            let participantId = parseInt($checkbox.data('participant-id'));
+            if (isNaN(participantId)) {
+                console.log('Error: Participant ID is not a number');
+                return;
+            }
+            $.getJSON('/menu/meal/' + participantId + '/isParticipationPending', function (isParticipationPendingResponse) {
+                if (isParticipationPendingResponse[0] === false) {
+                    ParticipationUpdateHandler.changeToOfferIsTaken($checkbox);
+                }
+            });
+        }
+
+        // no offers available (anymore)
+        if (data.isAvailable === false && $checkbox.hasClass(ParticipationAction.ACCEPT_OFFER) === true) {
+            ParticipationUpdateHandler.changeToOfferIsGone($checkbox);
+        }
+    }
+
+    private static handleUpdateSlots(data: SlotData)
+    {
+        const slotOption = $(`#day-${data.date}-slots option[value=${data.slotSlug}]`);
```

Comment:
can be writen like

$(`#day-${data.date}-slots option[value=${data.slotSlug}]`)
        .text(`${slotOption.data('title')} (${data.slotCount}/${slotOption.data('limit')})`)
        .prop('disabled', slotOption.data('limit') <= data.slotCount);